elastic/kibana
# [Synthetics] Omit or include `ssl` keys when appropriate for project monitors and private locations

## Summary

Resolves https://github.com/elastic/kibana/issues/149083

1. [Prevents tls fields from being saved](https://github.com/elastic/kibana/pull/149298/files#diff-56296f634bf379eb71629f426c670cd030d2a15263a59964847c0d10af09a767R14) on the Synthetics Integration policy when `is_tls_enabled` is false
2. Ensures `is_tls_enabled` is set properly for project monitors ([http](https://github.com/elastic/kibana/pull/149298/files#diff-0f42bb3b11a6ab864dee3488d5e9f7282adc009a261b3caee743a880b825c766R73) and [tcp](https://github.com/elastic/kibana/pull/149298/files#diff-3ad87e629abc6f17c395e8435c94f0f1a6274c9efea7d24ab81b7635ef0e43dfR69)). This ensures that when a monitor is sent to a public location or a private location, the `ssl` fields are sent or stripped appropriately.

### Testing

1. Create a private location
2. Create 2 lightweight project monitors using the following configuration
```
- type: tcp
  id: 'tls-enabled'
  name: 'TLS-Enabled'
  hosts: ["8.8.8.8:80"]
  ssl:
     verification_mode: 'strict'
```
```
- type: tcp
  id: 'tls-disabled'
  name: 'TLS-Disabled'
  hosts: ["8.8.8.8:80"]
```
3. Set these monitors to execute from both a private and public location via the `monitor` key in your `synthetics.config.ts` file. 
```
    monitor: {
      schedule: 3,
      privateLocations: ["YOUR PRIVATE LOCATION"],
      locations: ["us_central"], // to test against dev environment
    },
```
4. Navigate to the agent policy for the private location and inspect the full policy. Ensure the Synthetics policy on the agent package policy does not have `ssl` fields set for ssl disabled monitor. Ensure the `ssl` fields are set for the ssl enabled monitor.
